spec/integration/install_spec.cr

```diff
@@ -605,4 +605,36 @@ describe "install" do
       stdout.should contain(%(W: Shard "noshardyml" version (0.1.0) doesn't have a shard.yml file))
     end
   end
+
+  it "shows error when branch does not exist" do
+    metadata = {dependencies: {web: {git: git_url(:web), branch: "foo"}}}
+    with_shard(metadata) do
+      ex = expect_raises(FailedCommand) { run "shards install --no-color" }
+      ex.stdout.should contain(%(E: Could not find branch foo in the repository #{git_url(:web)} for shard "web"))
```

Comment:
I'd suggest to change the order of shard and url: 
```suggestion
      ex.stdout.should contain(%(E: Could not find branch foo for shard "web" in the repository #{git_url(:web)}))
```

The shard name is typically the more relevant information because that's where you look it up in the dependencies specification. Shard names are usually relatively short while git URLs can be pretty long, so it's easier to read that way.